src/python/grpcio/grpc/_server.py

```diff
@@ -764,7 +764,7 @@ def __init__(self, completion_queue, server, generic_handlers,
         self.interceptor_pipeline = interceptor_pipeline
         self.thread_pool = thread_pool
         self.stage = _ServerStage.STOPPED
-        self.shutdown_events = None
+        self.shutdown_events = []
```

Comment:
Change to:

```
self.termination_event = threading.Event()
self.shutdown_events = [self.termination_event]
```

Reply:
What's the use case you see in terms of giving applications access of the actual `Event` object?

Reply:
This is not giving applications access to event; it's just simplifying your implementation.

Reply:
That make sense. Changed.